src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -296,9 +307,9 @@ public void execute() {
             log("compiled on " + compileTimestamp, Project.MSG_VERBOSE);
 
             // Check for no arguments
-            if (fileName == null && fileSets.isEmpty()) {
+            if (fileName == null && fileSets.isEmpty() && sourcePaths.isEmpty()) {
```

Comment:
@MEZk @romani Is this how all out future conditions should be?
Should we make a new check in sevntu to enforce this behavior?